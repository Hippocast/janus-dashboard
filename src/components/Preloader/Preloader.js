import React from 'react';
import BubblePreloader from 'react-bubble-preloader';

import './Preloader.css';

const Preloader = () => (
    <BubblePreloader
        bubble={{ width: '2rem', height: '2rem' }}
        animation={{ speed: 2 }}
        className="j-preloader"
        colors={['#90c200', '#bde841', '#cee48f']}
    />
);

export default Preloader;
