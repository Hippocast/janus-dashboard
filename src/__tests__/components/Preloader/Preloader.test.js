import React from 'react';
import renderer from 'react-test-renderer';
import BubblePreloader from 'react-bubble-preloader';

import Preloader from '../../../components/Preloader/Preloader';

describe('Preloader component', () => {
    const passedProps = {
        discardPagination: jest.fn(),
        placeholder: 'Search...',
        searchQuery: '',
        setSearchQuery: jest.fn(),
    };

    it('renders correctly', () => {
        const tree = renderer
            .create(
                <Preloader />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
