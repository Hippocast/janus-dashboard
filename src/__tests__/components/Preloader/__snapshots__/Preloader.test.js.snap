// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader component renders correctly 1`] = `
<div
  className="bubble-loader j-preloader"
>
  <div
    className="bubble1"
    style={
      Object {
        "animation": "bouncedelay 2s ease-in-out infinite both",
        "animationDelay": "-0.32s",
        "backgroundColor": "#90c200",
        "borderRadius": "50%",
        "display": "inline-block",
        "height": "2rem",
        "width": "2rem",
      }
    }
  />
  <div
    className="bubble2"
    style={
      Object {
        "animation": "bouncedelay 2s ease-in-out infinite both",
        "animationDelay": "-0.16s",
        "backgroundColor": "#bde841",
        "borderRadius": "50%",
        "display": "inline-block",
        "height": "2rem",
        "width": "2rem",
      }
    }
  />
  <div
    className="bubble3"
    style={
      Object {
        "animation": "bouncedelay 2s ease-in-out infinite both",
        "backgroundColor": "#cee48f",
        "borderRadius": "50%",
        "display": "inline-block",
        "height": "2rem",
        "width": "2rem",
      }
    }
  />
</div>
`;
